packages/sdk/contractkit/src/wrappers/FeeCurrencyWhitelistWrapper.ts

```diff
@@ -0,0 +1,68 @@
+// NOTE: removing this import results in `yarn build` failures in Dockerfiles
+// after the move to node 10. This allows types to be inferred without
+// referencing '@celo/utils/node_modules/bignumber.js'
+import { FeeCurrencyWhitelist } from '@celo/abis/web3/FeeCurrencyWhitelist'
+import { StrongAddress } from '@celo/base'
+import 'bignumber.js'
+import { BaseWrapper, proxyCall } from './BaseWrapper'
+
+const minimal_token_info_abi = [
+  {
+    type: 'function' as const,
+    stateMutability: 'view',
+    outputs: [{ type: 'string', name: '', internalType: 'string' }],
+    name: 'symbol',
+    inputs: [],
+  },
+  {
+    type: 'function' as const,
+    stateMutability: 'view',
+    outputs: [{ type: 'string', name: '', internalType: 'string' }],
+    name: 'name',
+    inputs: [],
+  },
+] as const
+
+/**
+ * FeeCurrencyWhitelist contract listing available currencies usable to pay fees
+ */
+export class FeeCurrencyWhitelistWrapper extends BaseWrapper<FeeCurrencyWhitelist> {
+  getWhitelist = proxyCall(
+    this.contract.methods.getWhitelist,
+    undefined,
+    (addresses) => [...new Set(addresses)].sort() as StrongAddress[]
+  )
+
+  async getFeeCurrencyInformation(whitelist?: StrongAddress[]): Promise<
+    {
+      name: string | undefined
+      address: StrongAddress
+      symbol: string | undefined
+    }[]
+  > {
+    const feeCurrencies = whitelist ?? (await this.getWhitelist())
+
+    return Promise.all(
+      feeCurrencies.map(async (address) => {
```

Comment:
this has me think of [marianos suggestion]( https://github.com/celo-org/celo-monorepo/pull/10910#issuecomment-1912764602
) to try calling `getAdaptedToken` on [contract for whitelisted addresses](https://github.com/celo-org/celo-monorepo/blob/a83c7dee5b78c917f8990334f37386f5f5f4c07a/packages/protocol/contracts-0.8/stability/interfaces/IFeeCurrencyAdapter.sol) to see if they use an adapter or not.

we would then use this to display if the whitelist address was for the adapter or just the straight token

but the question is do we want to have this here or would that be doing to much?

Reply:
one option is leave as is and have a separate method which sees about the fee token adapter

Reply:
see [0625f93](https://github.com/celo-org/developer-tooling/pull/151/commits/0625f9364a8a1cb2d3303c77cccfdb754ff94685)